test/e2e/service.go

```diff
@@ -873,6 +873,103 @@ spec:
 		}()
 	})
 
+	ginkgo.It("Should ensure connectivity works on an external service when mtu changes in intermediate node", func() {
+		err := framework.WaitForServiceEndpointsNum(f.ClientSet, namespaceName, svcName, 4, time.Second, time.Second*180)
+		framework.ExpectNoError(err, fmt.Sprintf("service: %s never had an endpoint, err: %v", svcName, err))
+
+		time.Sleep(time.Second * 5) // buffer to ensure all rules are created correctly
+
+		endpoints, err := getEndpointsForService(f.ClientSet, namespaceName, svcName)
+		framework.ExpectNoError(err, fmt.Sprintf("failed to get endpoints for service %s", svcName))
+		gomega.Expect(endpoints).NotTo(gomega.BeNil())
+		gomega.Expect(len(endpoints.Subsets)).To(gomega.Equal(1))
+		gomega.Expect(len(endpoints.Subsets[0].Addresses)).To(gomega.Equal(4))
+		endPointIP := endpoints.Subsets[0].Addresses[0].IP
+		nodeName := endpoints.Subsets[0].Addresses[0].NodeName
+		nodeIP, err := getNodeIP(f.ClientSet, *nodeName)
+		framework.ExpectNoError(err, fmt.Sprintf("failed to get endpoint's %s node ip address", endPointIP))
+
+		// The external client configuration done in install_metallb can not be used because routes for external client
+		// installed in K8s node https://github.com/ovn-org/ovn-kubernetes/blob/master/contrib/kind.sh#L1045-L1047
+		// are ignored in shared gateway mode and traffic coming back from pod is put on the docker bridge directly by
+		// br-ex flows which needs to be handled in host(or vm) network.
+		// Hence the following set of ip commands set up two networks called bridge (192.168.222.0/24) and
+		// client 192.168.223.0/24 on the host(or vm) machine. The external client (192.168.223.2) runs on the client
+		// network which tries to connect with load balancer service via bridge network. There would be also a route
+		// created for the load balancer service forwarding packet into the node which hosts one of the endpoint.
```

Comment:
its hard to follow all of these ip commands and build the topology. Could you add an ascii diagram in here showing the topology with adjusted MTU and the connectivity the cluster?

Reply:
oh! yes, added a diagram now. hope its ok.

Reply:
looks good, thank you